kong/dao/schemas_validation.lua

```diff
@@ -95,9 +98,16 @@ function _M.validate_entity(tbl, schema, options)
       end
 
       -- Check the given table against a given schema
+
       for column, v in pairs(schema.fields) do
+        if not partial_update then
+          if t[column] == ngx.null then
+            t[column] = nil
```

Comment:
ditto: it seems the behavior we envision here is to treat fields set with `NULL` during a `PUT` as if they were initially created (run validation as if they were left blank, which means generates defaults and defaults for their sub-schema).

Could we have a couple of tests for this behavior?
- A test asserting that during a `full` update, fields set to `ngx.null` are reset to their `default`
- A test asserting that during a `full` update, fields with a sub-schema are reset to their `default`

You can create a fixture schema for this (the schema validation tests do it), in order to avoid depending on schemas from plugins or core entities :)